plugins/postgres/dbt/adapters/postgres/impl.py

```diff
@@ -21,64 +21,95 @@ class PostgresAdapter(SQLAdapter):
     def date_function(cls):
         return 'datenow()'
 
+    def _link_cached_database_relations(self, manifest, database, schemas):
+        table = self.execute_macro(manifest, GET_RELATIONS_MACRO_NAME)
+
+        for (refed_schema, refed_name, dep_schema, dep_name) in table:
+            referenced = self.Relation.create(
+                database=database,
+                schema=refed_schema,
+                identifier=refed_name
+            )
+            dependent = self.Relation.create(
+                database=database,
+                schema=dep_schema,
+                identifier=dep_name
+            )
+
+            # don't record in cache if this relation isn't in a relevant
+            # schema
+            if refed_schema.lower() in schemas:
+                self.cache.add_link(dependent, referenced)
+
     def _link_cached_relations(self, manifest):
         schemas = manifest.get_used_schemas()
+        # make a map of {db: [schemas]}
+        schema_map = {}
+        for db, schema in schemas:
+            schema_map.setdefault(db, []).append(schema.lower())
+
         try:
-            table = self.execute_macro(manifest, GET_RELATIONS_MACRO_NAME)
+            for db, schemas in schema_map.items():
+                self._link_cached_database_relations(manifest, db, schemas)
```

Comment:
this is called multiple times but it the `db` argument isn't passed to the get relations macro as far as I can tell.... are these calls duplicating work?

Reply:
On what adapter? On postgres/redshift we don't use the db argument in the sql because it's invalid (`{database}.information_schema` isn't accepted as a schema).